app/src/main/java/com/alexis/myanimecompanion/data/remote/MyAnimeListAPI.kt

```diff
@@ -0,0 +1,66 @@
+package com.alexis.myanimecompanion.data.remote
+
+import com.alexis.myanimecompanion.QueryFieldsBuilder
+import com.alexis.myanimecompanion.data.remote.models.*
+import retrofit2.http.*
+
+interface MyAnimeListAPI {
+    @GET("anime")
+    suspend fun search(
+        @Query("q") query: String? = null,
+        @Query("limit") limit: Int? = null,
+        @Query("offset") offset: Int? = null,
+        @Query("fields") fields: String? = null,
+    ): SearchResult
+
+    @GET("anime/{anime_id}")
+    suspend fun getAnimeDetails(
+        @Path("anime_id") animeId: Int,
+        @Query("fields") fields: String? = QueryFieldsBuilder.fieldsForAnimeDetails().done(),
+        @Header("Authentication") token: String? = null,
```

Comment:
Can this not be an optional parameter? It seems like token is always passed when getAnimeDetails() is being called

Reply:
I see. I'm having some doubts if there should be different method for logged Users or not. Please ignore for now